### PR TITLE
Add remote session discovery

### PR DESCRIPTION
**Pull request by dpwiz**

### CODE REVIEW

File: src/Simplex/Chat/Controller.hs [hunk 1]
```diff
@@ -906,7 +908,7 @@ data ChatError
   | ChatErrorAgent {agentError :: AgentErrorType, connectionEntity_ :: Maybe ConnectionEntity}
   | ChatErrorStore {storeError :: StoreError}
   | ChatErrorDatabase {databaseError :: DatabaseError}
-  | ChatErrorRemoteCtrl {remoteCtrlId :: RemoteCtrlId, remoteControllerError :: RemoteCtrlError}
+  | ChatErrorRemoteCtrl {remoteControllerError :: RemoteCtrlError}
```
Comment (epoberezkin) [inline src/Simplex/Chat/Controller.hs:911]:
```suggestion
  | ChatErrorRemoteCtrl {remoteCtrlError :: RemoteCtrlError}
```

File: src/Simplex/Chat/Migrations/M20230922_remote_controller.hs [hunk 2]
```diff
@@ -19,7 +19,8 @@ CREATE TABLE remote_hosts ( -- hosts known to a controlling app
 CREATE TABLE remote_controllers ( -- controllers known to a hosting app
   remote_controller_id INTEGER PRIMARY KEY,
   display_name TEXT NOT NULL,
-  fingerprint BLOB NOT NULL
+  fingerprint BLOB NOT NULL,
+  accepted INTEGER
```
Comment (epoberezkin) [inline src/Simplex/Chat/Migrations/M20230922_remote_controller.hs:23]:
```suggestion
  accepted INTEGER NOT NULL DEFAULT 0
```

File: src/Simplex/Chat/Remote/Discovery.hs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            TLS.serverShared = def {TLS.sharedCredentials = credentials},
+            TLS.serverHooks = def,
+            TLS.serverSupported = supportedParameters
+          }
+  httpClient <- newEmptyMVar
+  liftIO $ runTransportServer started partyPort serverParams defaultTransportServerConfig (run aPid httpClient)
+  takeMVar httpClient
+  where
+    announcer started inviteBS = do
+      atomically (takeTMVar started) >>= \case
```
Comment (epoberezkin) [inline src/Simplex/Chat/Remote/Discovery.hs:47]: ifM

File: src/Simplex/Chat/Migrations/M20230922_remote_controller.hs (hunk 2 above)
Comment (dpwiz) [reply src/Simplex/Chat/Migrations/M20230922_remote_controller.hs:23]: It's a tri-state field: non-resolved, resolved as accepted/rejected.

File: src/Simplex/Chat/Remote.hs [hunk 4]
```diff
@@ -90,3 +114,89 @@ sum2tagged :: J.Value -> J.Value
...
+        uiEvent <- async $ atomically $ readTMVar accepted
+        waitEitherCatchCancel listener uiEvent >>= \case
+          Left _ -> pure () -- discover got cancelled or crashed on some UDP error
+          Right (Left _) -> pure () -- readTMVar blocked indefinitely (should not happen)
+          Right (Right remoteCtrlId) -> do
+            -- got connection confirmation
+            (source, fingerprint) <-
+              atomically $
+                TM.lookup remoteCtrlId discovered >>= \case
+                  Nothing -> error "Session accepted without getting registered"
```
Comment (epoberezkin) [inline src/Simplex/Chat/Remote.hs:141]: shouldn't be error

**Pull request merged**